pkg/apis/core/v1/defaults_test.go

```diff
@@ -2052,3 +2052,58 @@ func TestSetDefaultServiceIPFamilyPolicy(t *testing.T) {
 
 	}
 }
+
+func TestSetDefaultServiceInternalTrafficPolicy(t *testing.T) {
+	all := v1.ServiceInternalTrafficPolicyAll
+	local := v1.ServiceInternalTrafficPolicyLocal
+	preferLocal := v1.ServiceInternalTrafficPolicyPreferLocal
+	testCases := []struct {
+		name                          string
+		expectedInternalTrafficPolicy v1.ServiceInternalTrafficPolicyType
+		svc                           v1.Service
+	}{
+		{
+			name:                          "must set default internalTrafficPolicy",
+			expectedInternalTrafficPolicy: v1.ServiceInternalTrafficPolicyAll,
+			svc:                           v1.Service{},
+		},
+		{
+			name:                          "must not set default internalTrafficPolicy when it's all",
+			expectedInternalTrafficPolicy: v1.ServiceInternalTrafficPolicyAll,
+			svc: v1.Service{
+				Spec: v1.ServiceSpec{
+					InternalTrafficPolicy: &all,
+				},
+			},
+		},
+		{
+			name:                          "must not set default internalTrafficPolicy when it's local",
+			expectedInternalTrafficPolicy: v1.ServiceInternalTrafficPolicyLocal,
+			svc: v1.Service{
+				Spec: v1.ServiceSpec{
+					InternalTrafficPolicy: &local,
+				},
+			},
+		},
+		{
+			name:                          "must not set default internalTrafficPolicy when it's prefer local",
+			expectedInternalTrafficPolicy: v1.ServiceInternalTrafficPolicyPreferLocal,
+			svc: v1.Service{
+				Spec: v1.ServiceSpec{
+					InternalTrafficPolicy: &preferLocal,
+				},
+			},
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceInternalTrafficPolicy, true)()
```

Comment:
include the feature gate bool in the test table and set it here so we can also test behavior when the gate is off